src/Hosting/Abstractions/src/IAppLifetime.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+
+namespace Microsoft.Extensions.Hosting
+{
+    /// <summary>
+    /// Allows consumers to perform cleanup during a graceful shutdown.
+    /// </summary>
+    public interface IAppLifetime
```

Comment:
Call it `IHostingApplicationLifetime` I hate the abbreviated App and it's anti the .NET naming guidelines

Reply:
IHostingApplicationLifetime's not great either. @glennc any ideas?

Reply:
Why is this interface duplicated again? It's identical right?

Reply:
So the name no longer conflicts with the one in Microsoft.AspNetCore.Hosting.

Reply:
How about ``IHostApplicationLifetime` so it at least mirrors the other new one and isn't quite as long?

Reply:
YUP.

Reply:
HostApplicationLifetime is what I expected but I don't remember why this has to be renamed and not moved. Do we not now have two identical things in two namespaces?